src/components/header/header.styles.ts

```diff
@@ -1,22 +1,27 @@
 import styled, {keyframes} from "styled-components";
 
-export const MainContainer = styled.div`
+interface OptionProps {
+  isLast?: boolean;  
+  hasImg?: boolean;  
```

Comment:
Acá los parametros de styledcomponents deberían ir con el prefijo $

```suggestion
  $isLast?: boolean;  
  $hasImg?: boolean;  
```

Sino react te tira un error

<img width="609" height="381" alt="Image" src="https://github.com/user-attachments/assets/91a57ed6-599e-47a4-8791-c5adbd8e0503" />

No solo vas a tener que cambiar la definición del nombre del parámetro sino también cuando lo usés